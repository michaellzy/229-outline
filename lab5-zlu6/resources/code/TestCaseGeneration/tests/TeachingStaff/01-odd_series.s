#---------------------------------------------------------------
# This simple program is used in Lab_BasicBlock to demonstrate
# how a Control Flow Graph is created.
#
# Register Usage:
#
#       a0: contains the number to be converted
#       a1: contains the address of the output buffer
#
#---------------------------------------------------------------

	.text
main:
	ori   $t0, $0, 0	     # i <-- 0
	ori   $v0, $0, 0# j <-- 0
	blez  $a0, DONE      # if (p <= 0) goto DONE
LOOP:	
	andi  $t2, $t0, 0x1  # $t2 <-- bit0 of i
	bne   $t2, $0,  ODD  # if i is odd goto ODD
	add   $v0, $v0, $t0  # j <-- j+i
        j     REINIT
ODD:	
	add   $v0, $v0, 1    # j <-- j+1
REINIT:
	add   $t0, $t0, 1    # i <-- i+1
	slt   $at, $t0, $a0
	bne   $at, $a0, LOOP # if i<p goto LOOP
DONE:	
	jr    $ra
