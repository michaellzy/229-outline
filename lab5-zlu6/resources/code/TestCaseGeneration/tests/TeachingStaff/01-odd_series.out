6 block(s) found.
Block Leader: 0x10010000, Size: 3
Block Leader: 0x1001000C, Size: 2
Block Leader: 0x10010014, Size: 2
Block Leader: 0x1001001C, Size: 1
Block Leader: 0x10010020, Size: 3
Block Leader: 0x1001002C, Size: 1

Edges:
0x10010000 --> 0x1001000C
0x10010000 --> 0x1001002C
0x1001000C --> 0x10010014
0x1001000C --> 0x1001001C
0x10010014 --> 0x10010020
0x1001001C --> 0x10010020
0x10010020 --> 0x1001000C
0x10010020 --> 0x1001002C

Dominator Bit Vectors:
0000 0000 0000 0000 0000 0000 0000 0001 
0000 0000 0000 0000 0000 0000 0000 0011 
0000 0000 0000 0000 0000 0000 0000 0111 
0000 0000 0000 0000 0000 0000 0000 1011 
0000 0000 0000 0000 0000 0000 0001 0011 
0000 0000 0000 0000 0000 0000 0010 0001 
